pkg/forwarder/forwarder.go

```diff
@@ -163,22 +175,35 @@ func New(cfg Config) (*Worker, error) {
 		if !pool.AppendCertsFromPEM(data) {
 			level.Warn(logger).Log("msg", "no certs found in from-ca-file")
 		}
-		fromTransport.TLSClientConfig.RootCAs = pool
-	}
-	fromClient := &http.Client{Transport: fromTransport}
-	if cfg.Debug {
-		fromClient.Transport = telemeterhttp.NewDebugRoundTripper(logger, fromClient.Transport)
-	}
-	if len(cfg.FromToken) == 0 && len(cfg.FromTokenFile) > 0 {
-		data, err := ioutil.ReadFile(cfg.FromTokenFile)
-		if err != nil {
-			return nil, fmt.Errorf("unable to read from-token-file: %v", err)
+
+		fromTransport.TLSClientConfig = &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      pool,
+		}
+
+		fromClient.Transport = telemeterhttp.NewTLSRoundTripper(fromClient.Transport)
+
+	} else {
```

Comment:
I would prefer to support multiple auth configurations at the same time rather being exclusive. E.g. if client cert/key files and token bearer are configured at the same time then setup the TLS configuration + the bearer round tripper. Same reasoning for the debug round-tripper, it could be useful even when using TLS cert/key only.

Reply:
done